alacritty/src/window.rs

```diff
@@ -153,15 +157,16 @@ impl Window {
         event_loop: &EventLoop<E>,
         config: &Config,
         size: Option<PhysicalSize<u32>>,
-        #[cfg(not(any(target_os = "macos", windows)))] wayland_event_queue: Option<&EventQueue>,
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+        wayland_event_queue: Option<&EventQueue>,
     ) -> Result<Window> {
         let window_config = &config.ui_config.window;
         let window_builder = Window::get_platform_window(&window_config.title, &window_config);
 
         // Disable vsync on Wayland.
-        #[cfg(not(any(target_os = "macos", windows)))]
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
         let vsync = !event_loop.is_wayland();
-        #[cfg(any(target_os = "macos", windows))]
+        #[cfg(any(target_os = "macos", windows, not(feature = "wayland")))]
         let vsync = true;
```

Comment:
How expensive is the Wayland check? We perform an is_wayland check here and later do an `is_x11` check. I think in theory we could just store this to a variable named x11 (since that seems our common name for this) and then use that for vsync and the checks later.

Would probably also help the optimizer.

Reply:
Well, there's no x11 on macOS/Windows and winit check is free, since it just checks enum variant and returns a constant.

Reply:
All we want to check is if it's Wayland or not, right? For everything else it should always use vsync? So I don't see what macOS/Windows have to do with it since we could just always define `x11 = false` in that case?

Reply:
updated.